test/utils.tests.js

```diff
@@ -147,13 +139,106 @@ describe('utils', () => {
+      assert.equal(result.erl_is_active_key, erlParams.erl_is_active_key);
+      assert.equal(result.erl_quota_key, erlParams.erl_quota_key);
+      assert.isFalse(result.erl_configured_for_bucket);
+    });
+    it('should return default keys when erlParams is undefined', () => {
+      const erlParams = undefined;
+      const bucketKeyConfig = {
+        size: 1,
+        interval: 60000,
+        per_interval: 1,
+        ttl: 60,
+        ms_per_interval: 0.000016666666666666667,
+        drip_interval: 60000
+      };
+      const result = resolveElevatedParams(erlParams, bucketKeyConfig);
+      assert.equal(result.ms_per_interval, bucketKeyConfig.ms_per_interval);
+      assert.equal(result.size, bucketKeyConfig.size);
+      assert.equal(result.erl_activation_period_seconds, 0);
+      assert.equal(result.erl_quota, 0);
+      assert.equal(result.erl_quota_interval, 'quota_per_calendar_month');
+      assert.equal(result.erl_is_active_key, 'defaultActiveKey');
+      assert.equal(result.erl_quota_key, 'defaultQuotaKey');
+      assert.isFalse(result.erl_configured_for_bucket);
+    });
+    it('should return appropriate keys when erlParam and bucketKeyConfig contain valid ERL values', () => {
+      const erlParams = {
+        erl_is_active_key: 'erl_is_active_key',
+        erl_quota_key: 'erl_quota_key',
+        erl_activation_period_seconds: 900,
+        erl_quota: 10,
+        erl_quota_interval: 'quota_per_calendar_month'
+      };
+      const bucketKeyConfig = {
+        size: 1,
+        interval: 60000,
+        per_interval: 1,
+        ttl: 60,
+        ms_per_interval: 0.000016666666666666667,
+        drip_interval: 60000,
+        elevated_limits: {
+          size: 2,
+          interval: 60000,
+          per_interval: 2,
+          ttl: 60,
+          ms_per_interval: 0.000033333333333333335,
+          drip_interval: 30000,
+          erl_configured_for_bucket: true,
+        }
+      };
+      const result = resolveElevatedParams(erlParams, bucketKeyConfig);
+      assert.equal(result.ms_per_interval, bucketKeyConfig.elevated_limits.ms_per_interval);
+      assert.equal(result.size, bucketKeyConfig.elevated_limits.size);
+      assert.equal(result.interval, bucketKeyConfig.elevated_limits.interval);
+      assert.equal(result.per_interval, bucketKeyConfig.elevated_limits.per_interval);
+      assert.equal(result.ttl, bucketKeyConfig.elevated_limits.ttl);
+      assert.equal(result.drip_interval, bucketKeyConfig.elevated_limits.drip_interval);
+      assert.equal(result.erl_activation_period_seconds, erlParams.erl_activation_period_seconds);
+      assert.equal(result.erl_quota, erlParams.erl_quota);
+      assert.equal(result.erl_quota_interval, erlParams.erl_quota_interval);
+      assert.equal(result.erl_is_active_key, erlParams.erl_is_active_key);
+      assert.equal(result.erl_quota_key, erlParams.erl_quota_key);
+      assert.equal(result.erl_configured_for_bucket, true);
     });
```

Comment:
We could split this to try to clarify the expectations
```suggestion
       describe('when erlParams and bucketKeyConfig contain valid ERL values', () => {
      const erlParams = {
        erl_is_active_key: 'erl_is_active_key',
        erl_quota_key: 'erl_quota_key',
        erl_activation_period_seconds: 900,
        erl_quota: 10,
        erl_quota_interval: 'quota_per_calendar_month'
      };
      const bucketKeyConfig = {
        size: 1,
        interval: 60000,
        per_interval: 1,
        ttl: 60,
        ms_per_interval: 0.000016666666666666667,
        drip_interval: 60000,
        elevated_limits: {
          size: 2,
          interval: 60000,
          per_interval: 2,
          ttl: 60,
          ms_per_interval: 0.000033333333333333335,
          drip_interval: 30000,
          erl_configured_for_bucket: true,
        }
      };
      
      const result = resolveElevatedParams(erlParams, bucketKeyConfig);
      
      it('should use the bucket details from bucketKeyConfig.elevated_limits', () => {
        assert.equal(result.ms_per_interval, bucketKeyConfig.elevated_limits.ms_per_interval);
        assert.equal(result.size, bucketKeyConfig.elevated_limits.size);
        assert.equal(result.interval, bucketKeyConfig.elevated_limits.interval);
        assert.equal(result.per_interval, bucketKeyConfig.elevated_limits.per_interval);
        assert.equal(result.ttl, bucketKeyConfig.elevated_limits.ttl);
        assert.equal(result.drip_interval, bucketKeyConfig.elevated_limits.drip_interval);
      });
      it('should use the quota, activation period and keys from erlParams', () => {
        assert.equal(result.erl_activation_period_seconds, erlParams.erl_activation_period_seconds);
        assert.equal(result.erl_quota, erlParams.erl_quota);
        assert.equal(result.erl_quota_interval, erlParams.erl_quota_interval);
        assert.equal(result.erl_is_active_key, erlParams.erl_is_active_key);
        assert.equal(result.erl_quota_key, erlParams.erl_quota_key);
        assert.equal(result.erl_configured_for_bucket, true);
      });
    })
```

Reply:
applied. thanks!